test/spmd/tensor/test_tensor_ops.py

```diff
@@ -95,6 +95,52 @@ def test_ones_like(self):
         ones_expected = torch.ones(4, 8)
         self.assertEqual(ones_expected, ones_like_dt.to_local())
 
+    @with_comms
+    def test_softmax(self):
+        device_mesh = DeviceMesh(
+            self.device_type, list(range(self.world_size))
+        )
+        x = torch.rand(8, 12, 16, device=self.device_type)
+        shard0_spec = Shard(0)
+        shard1_spec = Shard(1)
+        shard2_spec = Shard(2)
+        replica_spec = Replicate()
+        dist_x = distribute_tensor(x, device_mesh, [shard0_spec])
+
+        # TODO: add batching on dim 1, 2, and -1 (last dim) cases
+        for softmax_dim in [1, 2, -1]:
+            local_y = torch.nn.functional.softmax(
+                x, dim=softmax_dim, dtype=torch.float32
+            )
+            dist_y = torch.nn.functional.softmax(
+                dist_x, dim=softmax_dim, dtype=torch.float32
+            )
+            self.assertTrue(dist_y.placements[0].is_shard(dim=0))
+            dist_y = dist_y.redistribute(device_mesh, [replica_spec])
+            self.assertEqual(dist_y.to_local(), local_y)
+
+    @with_comms
+    def test_softmax_with_bwd(self):
+        device_mesh = DeviceMesh(self.device_type, list(range(self.world_size)))
+        x = torch.rand(8, 12, 16, device=self.device_type, requires_grad=True)
+        self.assertTrue(x.requires_grad)
+        local_y = x.softmax(dim=-1)
+        local_y.sum().backward()
+
+        shard0_spec = Shard(0)
+        dist_x = distribute_tensor(x, device_mesh, [shard0_spec])
+        self.assertTrue(dist_x.requires_grad)
+
+        dist_y = dist_x.softmax(dim=-1)
+        self.assertTrue(dist_y.placements[0].is_shard(dim=0))
+        dist_y_grad = torch.ones_like(dist_y)
+        # sum().backward() on dist_y has issue:
+        # dist_y.sum().backward(dist_y_grad)
+        # RuntimeError: Mismatch in shape: grad_output[0] has a shape of torch.Size([8, 12, 16]) and output[0] has a shape of torch.Size([]).
+        dist_y.backward(dist_y_grad)
+        self.assertIsNotNone(dist_x.grad)
```

Comment:
As we discussed, let's just use sum() for now. Thanks!

Reply:
> https://github.com/pytorch/tau/pull/440#discussion_r965352528
Is there a quick way to check? Because running `pytest test/spmd/tensor/test_dtensor_ops.py` takes a long time. Can I use `-s` option with this file? something like  `pytest test/spmd/tensor/test_dtensor_ops.py -s -k test_softmax`?